core/src/main/java/org/mule/api/registry/LifecycleRegistry.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.api.registry;
+
+import org.mule.api.MuleException;
+import org.mule.api.lifecycle.Injector;
+
+/**
+ * A {@link Registry} which not only registers and fetches objects,
+ * but it's also capable of applying lifecycle and injects dependencies
```

Comment:
The name still isn't great. LifecycleEnabled would be better, but that doesn't reflect that it also does injection.  Maybe *ManagedRegistry* or something similiar?

Be good to explain *when* it applies lifecycle.  Is it when an object is registered/unregistered or when LifecycleRegistry changes lifecycle state or both?

Also shouldn't this interface also extends Startable/Stoppable if it is going to apply lifecycle to the objects it had registered?